parastell/utils.py

```diff
@@ -56,124 +58,63 @@ def expand_ang_list(ang_list, num_ang):
-    'export_dir': ''
-}
-
-# dagmc_export_def (dict): dictionary of DAGMC export parameters including
-#     {
-#         'skip_imprint': choose whether to imprint and merge all in
-#             Coreform Cubit or to merge surfaces based on import order
-#             and geometry information (bool).
-#         'legacy_faceting': choose legacy or native faceting for DAGMC
-#             export (bool).
-#         'faceting_tolerance': maximum distance a facet may be from
-#             surface of CAD representation for DAGMC export (double).
-#         'length_tolerance': maximum length of facet edge for DAGMC
-#             export (double).
-#         'normal_tolerance': maximum change in angle between normal
-#             vector of adjacent facets (double).
-#         'anisotropic_ratio': controls edge length ratio of elements
-#             (double).
-#         'deviation_angle': controls deviation angle of facet from
-#             surface (i.e., lesser deviation angle results in more
-#             elements in areas with greater curvature) (double).
-#         'filename': name of DAGMC output file, excluding '.h5m'
-#             extension (str, defaults to 'dagmc').
-#         'export_dir': directory to which to export DAGMC output file
-#             (str, defaults to empty string).
-#             }
-dagmc_export_def = {
-    'skip_imprint': False,
-    'legacy_faceting': True,
-    'faceting_tolerance': None,
-    'length_tolerance': None,
-    'normal_tolerance': None,
-    'anisotropic_ratio': 100,
-    'deviation_angle': 5,
-    'filename': 'dagmc',
-    'export_dir': ''
-}
+
+def construct_kwargs_from_dict(
+    dict, allowed_kwargs, all_kwargs, fn_name=None, logger=None
+):
+    """Constructs a dictionary of keyword arguments with corresponding values
+    for a class method from an input dictionary based on a list of allowable
+    keyword argument argument names. Conditionally raises an exception if the
+    user flags that the supplied list of allowable arguments should represent
+    all keys present in the input dictionary of arguments.
+
+    Arguments:
+        dict (dict): dictionary of arguments and corresponding values.
+        allowed_kwargs (list of str): list of allowed keyword argument names.
+        all_kwargs (bool): flag to indicate whether 'allowed_kwargs' should
+            represent all keys present in 'dict'.
+        fn_name (str): name of class method (optional, defaults to None). If
+            'all_kwargs' is True, a method name should be supplied.
+        logger (object): logger object (optional, defaults to None). If
+            'all_kwargs' is True, a logger object should be supplied.
+
+    Returns:
+        kwarg_dict (dict): dictionary of keyword arguments and values.
+    """
+    kwarg_dict = {}
+    for name, value in dict.items():
+        if name in allowed_kwargs:
+            kwarg_dict.update({name: value})
+        elif all_kwargs and name not in allowed_kwargs:
```

Comment:
Isn't the second half of this condition implied by the fact that you didn't match on the `if`?

Reply:
Well I want the first condition to happen whether `all_kwargs` is `True` or `False`, while I want the error to be raised only under the condition that `all_kwargs = True`.

This functionality is particularly useful when extracting keyword arguments from YAML-derived dictionaries, since not all keys in those dictionaries are keyword arguments. In such cases, we set `all_kwargs = False` since we don't want an error to be raised when the algorithm comes across a key that is a positional argument.

Reply:
But I think you only get to this check against `all_kwargs` if `name not in allowed_kwargs`... right?
```suggestion
        elif all_kwargs:
```

Reply:
Oh, I see what you're saying. Good point! I thought the "second half of the condition" you were referring to was the `elif` as a whole.